tutorials/getting_started/so_you_built_a_model.md

```diff
@@ -0,0 +1,182 @@
+Your `DatasetReader` should already have a
+[`text_to_instance`](https://allenai.github.io/allennlp-docs/api/allennlp.data.dataset_readers.dataset_reader.html#allennlp.data.dataset_readers.dataset_reader.DatasetReader.text_to_instance)
+method, and so all you have to do is extract what it needs from the JSON.
+
+Accordingly, the existing Bidaf predictor is very simple:
+
+```python
+@Predictor.register('machine-comprehension')
+class BidafPredictor(Predictor):
+    """
+    Wrapper for the :class:`~allennlp.models.bidaf.BidirectionalAttentionFlow` model.
+    """
+    @overrides
+    def _json_to_instance(self, json_dict: JsonDict) -> Tuple[Instance, JsonDict]:
+        """
+        Expects JSON that looks like ``{"question": "...", "passage": "..."}``.
+        """
+        question_text = json_dict["question"]
+        passage_text = json_dict["passage"]
+        return self._dataset_reader.text_to_instance(question_text, passage_text), {}
+```
+
+As you can see, it just pulls the `"question"` and `"passage"` fields out of the input JSON
+and feeds them to `text_to_instance`. (The second return parameter is an output dict
+that the results of `Model.forward_on_instance` will get added to.  Here we use an empty dict,
+but if you wanted (for example) the `question_text` included in the output, you could make
+that happen here.)
+
+## Making Predictions
+
+As in the
+[Using AllenNLP in Your Project](https://github.com/allenai/allennlp/blob/master/tutorials/getting_started/using_in_your_repo.md#step-five-train-the-model) tutorial,
+we'll need to create a custom `run.py` script that imports our custom registered classes
+(that is, our `DatasetReader`, our `Model`, our `Predictor`, and anything else they rely on)
+and then [calls `main()`](https://github.com/allenai/allennlp/blob/master/allennlp/commands/__init__.py#L28).
+
+Here we have one additional step: we need to associate our custom predictor with our model.
+This is accomplished via the
+
+```python
+         predictor_overrides: Dict[str, str] = {},
+```
+
+parameter, which is a mapping from `Model` name to `Predictor` name.
+
+In our case we'd have to call
+
+```python
+    main(prog="python run.py",
+         predictor_overrides={'bidaf': 'machine-comprehension'})
+```
+
+to specify that when we see a `bidaf` model and need a `Predictor`
+we should use the `machine-comprehension` one.
+(As mentioned before, these are already associated
+[by default](https://github.com/allenai/allennlp/blob/master/allennlp/commands/predict.py#L35),
+but for tutorial reasons we're pretending they're not.)
+
+At this point you just need an input file in JSON lines format:
+
+```
+{"question": "First question...", "passage": "First passage..."}
+{"question": "Second question...", "passage": "Second passage..."}
+...
+```
+
+If you run your script, you should get a help message:
+
+```bash
+$ run.py predict
+usage: python -m allennlp.run predict [-h]
```

Comment:
This usage info is wrong - you've set `prog="python run.py"`, so you should get `usage: python run.py predict [-h]...`